---
title: "Class 13: Genome Informatics (part 1)"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Sample genotypes in the MXL 1000 Genome project data

Here we focus on the Mexican Ancestry in LA, CA

```{r}
#read CSV from ensemble
mxl <- read.csv("373531-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")
head(mxl)
```

How many of each genotype are there?
```{r}
table(mxl$Genotype..forward.strand.)
```

Proportion or percent of total for each genotype
```{r}
(table(mxl$Genotype..forward.strand.) / nrow(mxl)) * 100
```
## Quality Scores in FASTQ files

The forth line of a FASTQ sequence format file encodes the quality score that tells us how good the sequence at a given gosition is (i.e. how likely it is to be correct)
```{r}
library(seqinr)
library(gtools)
#phred <- asc(s2c("DDDDCDEDCDDDDBBDDDCC@"))

asc(s2c("DDDDCDEDCDDDDBBDDDCC@")) - 33
```

```{r}
f <- read.csv("rs8067378_ENSG00000172057.6.txt")
head(f)
```
##Population Scale Analysis

Read RNA-seq count data with genotype information results table

```{r}

```


```{r}
expr <- read.table("rs8067378_ENSG00000172057.6.txt")
head(expr)
```

```{r}
summary(expr)
```

```{r}
inds <- expr$geno == "G/G"
summary(expr[inds,"exp"])
```

```{r}
boxplot(exp ~ geno , data=expr)
```





